sourcegraph/sourcegraph
# code monitors: don't fail job if repo empty

Previously we failed a code monitor search if any of the revisions could not be resolved. For example, this happens if an empty repo is added to Sourcegraph. Now we log the error and continue with the job. The log message is shown in the Code Monitors log view.

<img width="1164" alt="image" src="https://github.com/sourcegraph/sourcegraph/assets/26413131/fe398e86-d995-43ca-8c3a-ece05b19a6f0">

Notes:
- The cm_trigger_jobs table had a field `log_contents: TEXT` which was unused. However, I anticipate we might want to add more logs in the future so `TEXT` is not the ideal type. Instead, I add a new column `logs: JSON[]`. This way we can add richer metadata to the logs if we want to.
- I have to drop `log_contents` in a separate PR, otherwise the backcompat tests fail.

### Test plan:
- manual testing
  - I added an empty repo to reproduce the error locally
  - I created a search job and verified that it passes with 0 results and that the logs are properly displayed in the UI (see screenshot)
- new unit test
